src/qofonoextcellinfo.cpp

```diff
@@ -268,6 +303,24 @@ void QOfonoExtCellInfo::Private::onCellsRemoved(QList<QDBusObjectPath> aCells)
     }
 }
 
+void QOfonoExtCellInfo::Private::onUnsubscribed()
+{
+    if (iActive) {
+        iActive = false;
+        Q_EMIT iParent->activeChanged();
+    }
+}
+
+void QOfonoExtCellInfo::Private::setActive(bool val)
+{
+    if (val != iActive) {
```

Comment:
Not sure how much a problem, but noting that this will not work if the state is flicked while an async call is being processed.